odl/test/space/npy_tensors_test.py

```diff
@@ -0,0 +1,1545 @@
+    assert x.norm() == pytest.approx(correct_norm)
+
+
+def test_norm_exceptions(tspace):
+    """Test if norm raises correctly for bad input."""
+    other_tspace = odl.rn((4, 3))
+    other_x = other_tspace.zero()
+
+    with pytest.raises(LinearSpaceTypeError):
+        tspace.norm(other_x)
+
+
+def test_pnorm(exponent):
+    """Test the norm method with p!=2 against numpy.linalg.norm."""
+    for tspace in (odl.rn((3, 4), exponent=exponent),
+                   odl.cn((3, 4), exponent=exponent)):
+        xarr, x = noise_elements(tspace)
+        correct_norm = np.linalg.norm(xarr.ravel(), ord=exponent)
+
+        assert tspace.norm(x) == pytest.approx(correct_norm)
+        assert x.norm() == pytest.approx(correct_norm)
+
+
+def test_dist(tspace):
+    """Test the dist method against numpy.linalg.norm of the difference."""
+    [xarr, yarr], [x, y] = noise_elements(tspace, n=2)
+
+    correct_dist = np.linalg.norm((xarr - yarr).ravel())
+    assert tspace.dist(x, y) == pytest.approx(correct_dist)
+    assert x.dist(y) == pytest.approx(correct_dist)
+
+
+def test_dist_exceptions(tspace):
+    """Test if dist raises correctly for bad input."""
+    other_tspace = odl.rn((4, 3))
+    other_x = other_tspace.zero()
+    x = tspace.zero()
+
+    with pytest.raises(LinearSpaceTypeError):
+        tspace.dist(other_x, x)
+
+    with pytest.raises(LinearSpaceTypeError):
+        tspace.dist(x, other_x)
+
+
+def test_pdist(exponent):
+    """Test the dist method with p!=2 against numpy.linalg.norm of diff."""
+    for tspace in (odl.rn((3, 4), exponent=exponent),
+                   odl.cn((3, 4), exponent=exponent)):
+        [xarr, yarr], [x, y] = noise_elements(tspace, n=2)
+
+        correct_dist = np.linalg.norm((xarr - yarr).ravel(), ord=exponent)
+        assert tspace.dist(x, y) == pytest.approx(correct_dist)
+        assert x.dist(y) == pytest.approx(correct_dist)
+
+
+def test_getitem(getitem_indices):
+    """Check if getitem produces correct values, shape and other stuff."""
+    tspace = odl.tensor_space((3, 4), dtype=complex, exponent=1, weighting=2)
+    x_arr, x = noise_elements(tspace)
+
+    x_arr_sliced = x_arr[getitem_indices]
+    sliced_shape = x_arr_sliced.shape
+    x_sliced = x[getitem_indices]
+
+    if np.isscalar(x_arr_sliced):
+        assert x_arr_sliced == x_sliced
+    else:
+        assert x_sliced.shape == sliced_shape
+        assert all_equal(x_sliced, x_arr_sliced)
+
+        # Check that the space properties are preserved
+        sliced_spc = x_sliced.space
+        assert sliced_spc.shape == sliced_shape
+        assert sliced_spc.dtype == tspace.dtype
+        assert sliced_spc.exponent == tspace.exponent
+        assert sliced_spc.weighting == tspace.weighting
+
+        # Check that we have a view that manipulates the original array
+        # (or not, depending on indexing style)
+        x_arr_sliced[:] = 0
+        x_sliced[:] = 0
+        assert all_equal(x_arr, x)
+
```

Comment:
My fault, I was thinking you modified both of them and would hence need to check both.